internal/provider/resource_assigned_managed_identity.go

```
@@ -0,0 +1,301 @@
+	_, err = t.client.ManagedIdentity.AssignManagedIdentityToWorkspace(ctx,
+		&ttypes.AssignManagedIdentityInput{
+			ManagedIdentityID: &managedIdentityID,
+			WorkspacePath:     workspace.FullPath,
+		})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating assigned managed identity",
+			err.Error(),
+		)
+		return
+	}
+
+	created := AssignedManagedIdentityModel{
+		ID:                types.StringValue(uuid.New().String()), // computed with no input from any other resource
+		ManagedIdentityID: types.StringValue(managedIdentityID),
+		WorkspaceID:       types.StringValue(workspace.Metadata.ID),
+	}
+
+	// Map the response body to the schema.  (There are no computed values to update to the plan.)
+	t.copyAssignedManagedIdentity(&created, &assignment)
+
+	// Set the response state to the fully-populated plan, whether or not there is an error.
+	resp.Diagnostics.Append(resp.State.Set(ctx, assignment)...)
+}
+
+func (t *assignedManagedIdentityResource) Read(ctx context.Context,
+	req resource.ReadRequest, resp *resource.ReadResponse,
+) {
+	// Get the current state.
+	var state AssignedManagedIdentityModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get the workspace from Tharsis so we can have its ID.
+	workspace, err := t.client.Workspaces.GetWorkspace(ctx,
+		&ttypes.GetWorkspaceInput{
+			ID: ptr.String(state.WorkspaceID.ValueString()),
+		},
+	)
+	if err != nil {
+		if tharsis.IsNotFoundError(err) {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+
+		resp.Diagnostics.AddError(
+			"Error reading workspace",
+			err.Error(),
+		)
+		return
+	}
+
+	// Get the assigned managed identities from Tharsis.
+	managedIdentities, err := t.client.Workspaces.GetAssignedManagedIdentities(ctx,
+		&ttypes.GetAssignedManagedIdentitiesInput{
+			ID: ptr.String(state.WorkspaceID.ValueString()),
+		})
+	if err != nil {
+		if tharsis.IsNotFoundError(err) {
+			resp.State.RemoveResource(ctx)
```

This should return an error since it indicates the workspace was not found.